### PR TITLE
Module5

### PR DESCRIPTION
**Pull request by JoLeeSoftwire**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+FROM python:3.8.5-buster as base
```
Comment (Jonesey13) [inline Dockerfile:1]: Is `FLASK_APP=todolist.app` in `.env.template` correct? Shouldn't it be `src.todolist.app`?

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+services:
+    todolist-prod:
+        build: 
+            context: .
+            target: development
+        env_file: ./.env
+        ports: 
+            - "5000:5000"
+        volumes:
+            - "/c/Work/DevOps/DevOps-Course-Starter/src:/todo-app/src"
```
Comment (Jonesey13) [inline docker-compose.yaml]: This won't work on everyone's machine.

This blocks this review from passing.

**Review by Jonesey13 — CHANGES_REQUESTED** [review]
This is most of the way there, but there are a couple more changes required:
* Adding container running instructions to the `README.md` file
* The docker compose file won't work on most machines

File: docker-compose.yaml (hunk 2 above)
Comment (JoLeeSoftwire) [reply docker-compose.yaml]: aha, oops, have made it a relative path

File: README.md [hunk 3]
```diff
@@ -2,13 +2,13 @@
...
-The project uses vagrant to run in a suitably configured VM. You will need Vagrant and VirtualBox installed.
+The project uses docker to run in a suitably configured container. You will need Docker installed.
 
-Copy the src/.env.template file to a new file `src/.env`, and populate it with your trello api credentials and board ID. 
+Copy the `.env.template` file to a new file `.env`, and populate it with your trello api credentials and board ID. 
 
 Then start the Flask app by running:
 ```bash
-$ vagrant up
+$ docker-compose up
```
Comment (hugh-emerson) [inline README.md]: You shouldn't really *require* developers to use docker - so you should have instructions to run `poetry install`/`poetry run flask run` and the alternatives of `vagrant up` / `docker-compose up`

Also, FYI this command will not rebuild the image even if you edit the docker file. Safer to use `docker-compose up --build`

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+version: "3.8"
+
+services:
+    todolist-prod:
```
Comment (hugh-emerson) [inline docker-compose.yaml]: Service name seems like a mistake

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+COPY pyproject.toml .
+COPY poetry.lock .
+RUN poetry config virtualenvs.create false --local
+RUN poetry install
+
+FROM base as production
+# Configure for production
+ENTRYPOINT [ "poetry" ]
+CMD ["run", "gunicorn", "--bind=0.0.0.0", "src.todolist.app:create_app()"]
+# run command: docker run --env-file .env -p 5000:8000 --mount type=bind,source="$(pwd)"/src,target=/todo-app/src todolist:prod
```
Comment (hugh-emerson) [inline Dockerfile]: - You shouldn't be mounting files for the production image - you should `COPY` the files over
- This command is implicitly dependent on first building the image with the tag todolist:prod

**Review by hugh-emerson — APPROVED** [review]
I've left a couple of suggestions but you can merge this when you're ready.

File: README.md (hunk 3 above)
Comment (JoLeeSoftwire) [reply README.md]: have added --build, running locally was covered further down the readme, I've added the vagrant instructions back in

**Pull request merged**